### PR TITLE
[Keymap] My ErgoDash Keymap

### PR DESCRIPTION
**Pull request by totuus1157**
## Description

My personal layout for ErgoDash

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the ### ### documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ergodash/rev1/keymaps/totuus1157/keymap.c [hunk 1]
```diff
@@ -0,0 +1,177 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/totuus1157/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/ergodash/rev1/keymaps/totuus1157/keymap.c [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
+#define _ADJUST 16
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/totuus1157/keymap.c:7]: We recommend using an enum for layer names, as they auto increment.  Especially in this case, as you're adding 13 unused layers, by setting adjust to layer 16.
```suggestion
enum custom_layers {
    _QWERTY,
    _LOWER,
    _RAISE,
    _ADJUST,
};
```

File: keyboards/ergodash/rev1/keymaps/totuus1157/keymap.c [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+};
+
+#ifdef AUDIO_ENABLE
+float tone_qwerty[][2]     = SONG(QWERTY_SOUND);
+#endif
+
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/totuus1157/keymap.c:136]: Could you remove this function?  

It's been deprecated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer
```suggestion
```

File: keyboards/ergodash/rev1/keymaps/totuus1157/keymap.c [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+         print("mode just switched to qwerty and this is a huge string\n");
+        set_single_persistent_default_layer(_QWERTY);
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/totuus1157/keymap.c:143]: If this is the only default layer, then you can remove this macro completely.

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by totuus1157** [comment]
Since we don't want to make it far from the default settings, we decided to drop this pull request.

**Pull request closed without merging**